src/software/ai/hl/stp/play/penalty_kick_enemy_play.cpp

```diff
@@ -19,19 +19,33 @@ bool PenaltyKickEnemyPlay::isApplicable(const World &world) const
 
 bool PenaltyKickEnemyPlay::invariantHolds(const World &world) const
 {
-    return world.gameState().isTheirPenalty();
+    return world.gameState().isTheirPenalty() && !world.gameState().isStopped() &&
+           !world.gameState().isHalted();
 }
 
 void PenaltyKickEnemyPlay::getNextTactics(TacticCoroutine::push_type &yield)
 {
     auto goalie_tactic = std::make_shared<GoalieTactic>(
         world.ball(), world.field(), world.friendlyTeam(), world.enemyTeam());
+    goalie_tactic->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    goalie_tactic->addWhitelistedAvoidArea(AvoidArea::BALL);
+    goalie_tactic->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_DEFENSE_AREA);
 
     auto move_tactic_2 = std::make_shared<MoveTactic>(true);
+    move_tactic_2->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    move_tactic_2->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
     auto move_tactic_3 = std::make_shared<MoveTactic>(true);
+    move_tactic_3->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    move_tactic_3->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
     auto move_tactic_4 = std::make_shared<MoveTactic>(true);
+    move_tactic_4->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    move_tactic_4->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
     auto move_tactic_5 = std::make_shared<MoveTactic>(true);
+    move_tactic_5->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    move_tactic_5->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
     auto move_tactic_6 = std::make_shared<MoveTactic>(true);
+    move_tactic_6->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    move_tactic_6->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
```

Comment:
We're allowing all the friendly robots into the friendly half when it's the enemies penalty kick? Shouldn't we restrict them to the enemy half?

Reply:
Yeah, that was just done because the robots had weird behavior when they got booted out of the friendly half. Our half was whitelisted so they would move nicely.

If they were deep into the enemy half when the play activated they would immediately go to the sidelines and awkwardly go up the side of the field out of bounds.

Reply:
Ah. I think the correct thing here would be to have the `ENEMY_HALF` `AvoidArea` actually go to +/- infinity in `y`, since semantically what we want to happen is the robots to avoid the enemy half (including sidelines). @MathewMacDougall - if you think this is reasonable, could you file a ticket to make the the change to the `AvoidArea::ENEMY_HALF` and remove the `ENEMY_HALF` whitelists here?

Reply:
I created #980, but I'm going to leave the AvoidAreas as they are for now so that we don't intentionally break behavior. I've left a comment linking to the ticket so we know to remove them later once that's resolved.